python/tank_vendor/shotgun_authentication/login_dialog.py

```diff
@@ -177,16 +198,74 @@ def _ok_pressed(self):
             QtGui.QApplication.processEvents()
 
             # try and authenticate
-            self._new_session_token = session_cache.generate_session_token(
+            token = session_cache.generate_session_token(
                 site, login, password, self._http_proxy
             )
         except AuthenticationError, e:
             # authentication did not succeed
-            self._set_error_message(e[0])
-            self.ui.message.show()
+            self._set_error_message(self.ui.message, e[0])
             return
```

Comment:
I think the code might be a little bit clearer if we just get rid of this return.